psi4/src/psi4/libfunctional/LibXCfunctional.cc

```diff
@@ -254,100 +253,45 @@ std::map<std::string, double> LibXCFunctional::query_libxc(const std::string& fu
-            failed = false;
-        }
-    } else if (xc_func_name_ == "XC_GGA_C_PBE") {
-        if (vsize == 1) {
-            // (XC(func_type) *p, FLOAT beta);
-            xc_gga_c_pbe_set_params(xc_functional_.get(), values[0]);
-            failed = false;
-        }
-    } else if (xc_func_name_ == "XC_GGA_X_PW91") {
-        if (vsize == 7) {
-            // (XC(func_type) *p, FLOAT a, FLOAT b, FLOAT c, FLOAT d, FLOAT f, FLOAT alpha, FLOAT expo);
-            xc_gga_x_pw91_set_params(xc_functional_.get(), values[0], values[1], values[2], values[3], values[4],
-                                     values[5], values[6]);
-            failed = false;
-        }
-    } else if (xc_func_name_ == "XC_GGA_X_RPBE") {
-        if (vsize == 2) {
-            // (XC(func_type) *p, FLOAT kappa, FLOAT mu);
-            xc_gga_x_rpbe_set_params(xc_functional_.get(), values[0], values[1]);
-            failed = false;
-        }
-    } else if (xc_func_name_ == "XC_GGA_X_OPTX") {
-        if (vsize == 3) {
-            // (XC(func_type) *p, FLOAT a, FLOAT b, FLOAT gamma);
-            xc_gga_x_optx_set_params(xc_functional_.get(), values[0], values[1], values[2]);
-            failed = false;
-        }
-    } else if (xc_func_name_ == "XC_GGA_C_LYP") {
-        if (vsize == 4) {
-            // (XC(func_type) *p, FLOAT A, FLOAT B, FLOAT c, FLOAT d);
-            xc_gga_c_lyp_set_params(xc_functional_.get(), values[0], values[1], values[2], values[3]);
-            failed = false;
-        }
-    } else if ((xc_func_name_ == "XC_HYB_GGA_XC_HSE03") || (xc_func_name_ == "XC_HYB_GGA_XC_HSE06")) {
+    int npars = xc_func_info_get_n_ext_params(xc_functional_.get()->info);
+    if (npars == 0) {
+        throw PSIEXCEPTION(
+            "LibXCfunctional: set_tweak: There are no known tweaks for this functional, please double check "
+            "the functional form and add them if required.");
+    } else if (npars != vsize) {
+        std::ostringstream oss;
+        oss << "got " << vsize << ", expected " << npars;
+        throw PSIEXCEPTION(
+            "LibXCfunctional: set_tweak: Mismatch in size of tweaker vector and expected number of "
+            "input parameters:" +
+            oss.str() + "\n");
+    }
+
+    if (xc_func_name_ == "XC_GGA_C_PBE") {
         if (vsize == 3) {
-            // "Mixing parameter beta", "Screening parameter omega_HF", "Screening parameter omega_PBE"
+            // (XC(func_type) *p, FLOAT beta); FLOAT gamma, FLOAT BB
+            // xc_gga_c_pbe_set_params(xc_functional_.get(), values[0]);
+            values[1] =
+                xc_func_info_get_ext_params_default_value(const_cast<xc_func_info_type*>(xc_functional_->info), 1);
+            values[2] =
+                xc_func_info_get_ext_params_default_value(const_cast<xc_func_info_type*>(xc_functional_->info), 2);
             xc_func_set_ext_params(xc_functional_.get(), values.data());
```

Comment:
Btw this will prohibit anyone from fully customizing `GGA_C_PBE`. I would recommend either putting the default values in the code, or changing the tweaker to only set the parameters you want to change using `void  xc_func_set_ext_params_name(xc_func_type *p, const char *name, double par);`